### PR TITLE
vscode-extensions.kalebpace.balena-vscode: init at 0.1.3

### PR DESCRIPTION
**Pull request by kalebpace**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by superherointj** [comment]
Certify to do unrelated refactor in a different commit.

**Comment by superherointj** [comment]
Meta fields are missing.

**Comment by superherointj** [comment]
Please fix:

> pkgs/applications/editors/vscode/extensions/default.nix:
> Line	1634: Trailing whitespace

(warn by editor config)

**Comment by kalebpace** [comment]
Hey superherointj, thanks for looking at this, appreciate your time. Unintentionally formatted/"refactors" the entire file when trying to address the editorconfig warning earlier. 

Let me know if there's anything else you'd like me to add or address!

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 1]
```diff
@@ -1632,6 +1632,21 @@ let
...
 
+      kalebpace.balena-vscode = buildVscodeMarketplaceExtension {
+        mktplcRef = {
+          name = "balena-vscode";
+          publisher = "kalebpace";
+          version = "0.1.3";
+          sha256 = "sha256-CecEv19nEtnMe0KlCMNBM9ZAjbAVgPNUcZ6cBxHw44M=";
+        };
+        meta = with lib; {
+          description = "Unofficial VS Code extension for use with Balena Cloud";
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/default.nix:1643]:
```suggestion
          description = "A VS Code extension for integration with Balena";
```

Comment (superherointj) [reply pkgs/applications/editors/vscode/extensions/default.nix:1643]: `changelog` and `downloadPage` are missing in meta.
Please copy from other examples in file as reference.

**Comment by superherointj** [comment]
> Hey superherointj, thanks for looking at this, appreciate your time.

S2

> Unintentionally formatted/"refactored" the entire file when trying to address the editorconfig warning earlier.

Yes, the file needs formatting. Feel free to file a PR fixing that, once we are done here.

> Let me know if there's anything else you'd like me to add or address!

Sorry about not leaving all requests in a single batch.

Thanks for your understanding.

**Comment by superherointj** [comment]
> error: undefined variable 'kalebpace'

You have to add yourself to maintainers file:
https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix

Also, certify to build this package on every change:

> nix build .#vscode-extensions.kalebpace.balena-vscode

Or run:
  - nix-shell -p nixpkgs-review
  - nixpkgs-review pr 221727

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 221727` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vscode-extensions.kalebpace.balena-vscode</li>
  </ul>
</details>

**Review by superherointj — APPROVED** [review]
@kalebpace Congratulations for your first contribution!

**Comment by kalebpace** [comment]
Thank you so much!

**Comment by superherointj** [comment]
> Thank you so much!

You are welcome.

Merge will happen after CI is done.

**Pull request merged**